### PR TITLE
User-definable startup melody

### PR DESCRIPTION
**Pull request by Rejdukien**
https://discordapp.com/channels/596350022191415318/596350022191415320/780478873258295296
Ask and you shall receive a frivolous feature.

Let's the user define a custom startup melody, kind of similar to BlHeli32.
Syntax: {Note, Value}, value being 1 for whole notes, 2 for half notes and so on.
The helper file pitches.h contains all the frequencies for various notes.

As it is, it should be the ff7 victory fanfare, more or less.

### CODE REVIEW

**Comment by maybenikhil** [comment]
to keep user-defines tidy, I think we should remove the comment in user_defines.txt. Also, would it be possible to add compatibility with BLHeli32 pastes or would that be to complex, because of the sharps (#) in the editor being recognized as comments.

File: src/user_defines.txt [hunk 1]
```diff
@@ -56,6 +56,10 @@
 
 #-DUSE_ESP8266_BACKPACK
 
+### uncomment MELODY_EDITOR and MELODY_EDITOR_BPM to specify your own startup jingle instead of the default
+### see pitches.h for valid note names
 #-DJUST_BEEP_ONCE
+-DMELODY_EDITOR="{{NOTE_G5, 16}, {NOTE_REST, 16}, {NOTE_G5, 16}, {NOTE_REST, 16}, {NOTE_G5, 16}, {NOTE_REST, 16}, {NOTE_G5, 2}, {NOTE_DS5, 2}, {NOTE_F5, 2}, {NOTE_G5, 8}, {NOTE_REST, 4}, {NOTE_F5, 8}, {NOTE_G5, 1}}"
```
Comment (JyeSmith) [inline src/user_defines.txt:62]: These should be commented by default, or changed to the current start up.

**Comment by JyeSmith** [comment]
Maybe have the ff7 victory fanfare, and other options, added to the wiki user defines for others to copy and paste.

**Comment by maybenikhil** [comment]
That totally works! I can just steal a bunch of Rox Wolf's Blheli32 tones and just edit the #'s into S's

**Comment by Rejdukien** [comment]
Straight up compatibility with BLHeli32 pastes (with the only change being replacing sharps with S's) would be great, I agree on that. Perhaps with a python preprocessor script to parse that string instead of defining an array as macro in user-defines. I'll see what I can do later!

One word of caution though, I've been experimenting a bit with that - They usually need to be transposed down a couple semitones or it sounds too high pitched.

**Review by cruwaller — APPROVED** [review]

**Comment by Rejdukien** [comment]
> That totally works! I can just steal a bunch of Rox Wolf's Blheli32 tones and just edit the #'s into S's

Alright, now you can steal all of them. 😂 

Slight difference, you can append "|<number>" to pitch everything up or down a number of semitones. For example, the default (commented out) melody is being pitched down by 3 semitones.
The only thing I couldn't figure out is what "Gen. Length" and "Gen. Interval" do exactly, so I've just stuck to the "bpm" variant of defining the tempo.

**Comment by JyeSmith** [comment]
Could MY_STARTUP_MELODY_BPM also be stuck on the end `|BPM`?  It just makes everything a single liner and neater.

**Comment by AlessandroAU** [comment]
Is this ready for merge? 👍

**Comment by Rejdukien** [comment]
Pretty much. Wouldn't know what else to add.

**Comment by AlessandroAU** [comment]
Awesome :) Thanks for the great work

**Pull request merged**